MyHashMap.java

```diff
@@ -0,0 +1,83 @@
+// Time Complexity :  O(1) amortized. worst case would be O(n)
+// Space Complexity : O(n)
+// Did this code successfully run on Leetcode : yes
+// Any problem you faced while coding this :  on line 66, was comparing 'prev.next.val != key' and spent some time 
```

Comment:
That is a neat observation. Please do spend more time visualizing the problem on the whiteboard.